src/query/functions/tests/it/scalars/testdata/array.txt

```diff
@@ -593,22 +582,30 @@ output domain  : [{FALSE, TRUE}]
 output         : [false, true]
 
 
-ast            : array_concat([false, true], [1,2])
-raw expr       : array_concat(array(false, true), array(1_u8, 2_u8))
-checked expr   : array_concat<T0=Variant><Array(T0), Array(T0)>(CAST(array<T0=Boolean><T0, T0>(false, true) AS Array(Variant)), CAST(array<T0=UInt8><T0, T0>(1_u8, 2_u8) AS Array(Variant)))
-optimized expr : [false, true, 1, 2]
-output type    : Array(Variant)
-output domain  : [Undefined]
-output         : [false, true, 1, 2]
+error: 
+  --> SQL:1:1
+  |
+1 | array_concat([false, true], [1,2])
```

Comment:
It is better to change the test case data to the same type rather than report an error.